core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -761,8 +761,9 @@ class SparkContext(config: SparkConf) extends Logging {
    */
   def setJobGroup(groupId: String,
       description: String, interruptOnCancel: Boolean = false): Unit = {
+    val actualGroupId = getJobGroupId(groupId)
```

Comment:
https://github.com/apache/spark/pull/31227/files#diff-1e596e9e9bd50e3075847a66f8fecee741289b2b0478f9099c3bf2b680ebaa35R749-R755

This looks like a silent API behavior change.